buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -1312,5 +1316,24 @@ private static boolean isUtf8(ByteBuf buf, int index, int length) {
         return true;
     }
 
+    public static int lengthOfByteOrderMark(ByteBuf buf, Charset charset) {
```

Comment:
can you clarify what this function does?   If the BOM is not found, this can return both 0 and 1.   Should it return -1 instead?    

Also, public methods should have a javadoc.

Reply:
I have added Javadoc with explanation.  As of now, this method is only used from the test code I added (`DelimitedLengthFieldBasedFrameDecoderTest`), so if desired I can refactor to only keep it there.  This was the place where it seemed like it should live (given my limited exposure to the codebase).

Updated the behavior to return -1 if BOM is not found at the beginning of the buffer.